practice-app/server/src/routes/chess/chess.controller.js

```diff
@@ -23,25 +27,145 @@ const ChessController = {
-                response.id,
-                selectedColor
-            );
-            if (game) {
-                return res.status(200).json({
+            if (response && response.id) {
+                const game = await ChessGame.create({
                     game_id: response.id,
                     player_color: selectedColor,
                 });
+                if (game) {
+                    return res.status(200).json({
+                        game_id: response.id,
+                        player_color: selectedColor,
+                    });
+                }
             }
+        } catch (e) {
+            console.log(e);
         }
 
         return res.status(500).json({
             message: "Could not initiate a game against the AI",
         });
     },
+    makeMove: async function (req, res) {
+        // Get user input
+        const { gameId, moveStr } = req.body;
+
+        const url = `https://lichess.org/api/board/game/${gameId}/move/${moveStr}`;
+
+        const payload = {};
+        const headers = {
+            headers: {
+                Authorization: `Bearer ${process.env.LICHESS_TOKEN}`,
+            },
+        };
+
+        try {
+            const response = (await axios.post(url, payload, headers)).data;
+
+            if (response && response.ok) {
+                return res.status(200).json({
+                    ok: true,
+                });
+            }
+        } catch (e) {
+            return res
+                .status(400)
+                .json({ message: "Could not make the move." });
+        }
+
+        return res.status(500).json(response);
+    },
+    streamGame: async function (req, res) {
+        const { gameId } = req.params;
+
+        const url = `https://lichess.org/api/board/game/stream/${gameId}`;
+        const headers = {
+            Authorization: `Bearer ${process.env.LICHESS_TOKEN}`,
+        };
+
+        try {
+            const stream = needle.get(url, {
+                headers,
+            });
+            stream
+                .pipe(ndjson.parse())
+                .on("data", (data) => {
+                    try {
+                        const state = data.state ? data.state : data;
+
+                        if (
+                            state.moves !== undefined &&
+                            state.status !== undefined
+                        ) {
+                            db.collection("chessgames").updateOne(
+                                { game_id: gameId },
+                                {
+                                    $set: {
+                                        moves: state.moves,
+                                        status: state.status,
+                                    },
+                                }
+                            );
+                        }
+
+                     
+                        if (state.status === "mate") {
+                            db.collection("chessgames").updateOne(
```

Comment:
You can use `ChessGame.updateOne` instead of `db.collection("chessgames").updateOne`. It is less error prone since you do not have to write collection name each time. Same issue in line 104

Reply:
changed to ChessGame.updateOne, removed db.js